segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/chunklayer/ChunkedSegmentStorage.java

```diff
@@ -191,14 +196,22 @@ public ChunkedSegmentStorage(int containerId, ChunkStorage chunkStorage, ChunkMe
      * Initializes the ChunkedSegmentStorage and bootstrap the metadata about storage metadata segments by reading and processing the journal.
      *
      * @param snapshotInfoStore Store that saves {@link SnapshotInfo}.
+     * @param taskQueue  Task queue to use for garbage collection.
      */
-    public CompletableFuture<Void> bootstrap(SnapshotInfoStore snapshotInfoStore) {
+    public CompletableFuture<Void> bootstrap(SnapshotInfoStore snapshotInfoStore, AbstractTaskQueueManager<GarbageCollector.TaskInfo> taskQueue) {
 
         this.logPrefix = String.format("ChunkedSegmentStorage[%d]", containerId);
-
+        this.taskQueue = taskQueue;
```

Comment:
You have a possible leak here. `taskQueue` is created in `StreamSegmentContainer` and is passed here via this method. However neither `StreamSegmentContainer` nor this class ever close it.

Normally it is up to whomever creates the object to manage its lifecycle, so I would have argued that the container should have closed it. However in this case, I think it's better if this class does it; if you choose to go this way, please clearly indicate in this argument's Javadoc what's going to happen to it once the `ChunkedSegmentStorage` class is closed.

Reply:
Fixed. Closing it with ChunkedSegmentStorage::close